subsys/bluetooth/audio/has.c

```diff
@@ -541,8 +619,61 @@ int bt_has_preset_unregister(uint8_t index)
 		return -ENOENT;
 	}
 
+	preset_changed_prepare(&buf, BT_HAS_CHANGE_ID_PRESET_DELETED, BT_HAS_IS_LAST);
+	net_buf_simple_add_u8(&buf, preset->index);
+
 	preset_free(preset);
 
+	return control_point_send_all(&buf);
+}
+
+int bt_has_preset_available(uint8_t index)
+{
+	struct has_preset *preset = NULL;
+
+	CHECKIF(index == BT_HAS_PRESET_INDEX_NONE) {
+		BT_ERR("index is invalid");
+		return -EINVAL;
+	}
+
+	/* toggle property bit if needed */
+	if (!(preset->properties & BT_HAS_PROP_AVAILABLE)) {
+		NET_BUF_SIMPLE_DEFINE(buf, sizeof(struct bt_has_cp_hdr) +
+				      sizeof(struct bt_has_cp_preset_changed) + sizeof(uint8_t));
+
+		preset->properties ^= BT_HAS_PROP_AVAILABLE;
+
+		preset_changed_prepare(&buf, BT_HAS_CHANGE_ID_PRESET_AVAILABLE, BT_HAS_IS_LAST);
+		net_buf_simple_add_u8(&buf, preset->index);
+
+		return control_point_send_all(&buf);
+	}
+
+	return 0;
+}
+
+int bt_has_preset_unavailable(uint8_t index)
+{
+	struct has_preset *preset = NULL;
+
+	CHECKIF(index == BT_HAS_PRESET_INDEX_NONE) {
+		BT_ERR("index is invalid");
```

Comment:
Same here. Btw, is this a normal situation that might occur in a bug free application, or does this indicate a bug? If the latter, then would an assert make more sense?

Reply:
I guess since you’ve exposed this in the shell, it’s not so nice to be able to trigger an assert eg with a typo.

Reply:
`CHECKIF` can be configured to do an assert if the developer would prefer that over an error code